src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -31,6 +33,21 @@ namespace Microsoft.PowerShell.EditorServices
     /// </summary>
     public class PowerShellContext : IDisposable, IHostSupportsInteractiveSession
     {
+        private const string DotNetFrameworkDescription = ".NET Framework";
+
+        private static readonly Action<Runspace, ApartmentState> s_runspaceApartmentStateSetter;
+
+        static PowerShellContext()
+        {
+            // PowerShell ApartmentState APIs aren't available in PSStandard, so we need to use reflection
+            if (RuntimeInformation.FrameworkDescription == DotNetFrameworkDescription)
+            {
+                MethodInfo setterInfo = typeof(Runspace).GetProperty("ApartmentState").GetSetMethod();
+                var setter = Delegate.CreateDelegate(typeof(Action<Runspace, ApartmentState>), firstArgument: null, method: setterInfo);
+                s_runspaceApartmentStateSetter = (Action<Runspace, ApartmentState>)setter;
```

Comment:
Neat! I always thought you needed an instance to create a delegate like that.  I'd have built out an expression for no reason 😄

Reply:
As always, I was fishing around in the darkness with reflection and PowerShell made life a lot easier for me prototyping it all.

Reply:
I know right! The true magic of PowerShell is rapid prototyping 😁